VideoLocker/src/main/java/org/edx/mobile/view/custom/MenuPopupHelper.java

```diff
@@ -0,0 +1,455 @@
+            return 2;
+        }
+
+        @Override
+        public int getItemViewType(int position) {
+            return getItemType(position).ordinal();
+        }
+
+        private ItemType getItemType(int position) {
+            int index = 0;
+            MenuItem expandedItem = mMenu.getExpandedItem();
+            for (MenuItem item : getMenuItems()) {
+                if (item != expandedItem) {
+                    Menu subMenu = item.getSubMenu();
+                    if (index++ == position) {
+                        return subMenu == null ?
+                                ItemType.ITEM : ItemType.HEADER;
+                    }
+                    if (subMenu != null) {
+                        int subMenuCount = subMenu.size();
+                        if (position < index + subMenuCount) {
+                            return ItemType.ITEM;
+                        }
+                        index += subMenuCount;
+                    }
+                }
+            }
+            throw new IndexOutOfBoundsException();
+        }
+
+        @Override
+        public boolean areAllItemsEnabled() {
+            return false;
+        }
+
+        @Override
+        public boolean isEnabled(int position) {
+            return getItemType(position) != ItemType.HEADER &&
+                    getItem(position).isEnabled();
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            TextView textView;
+            if (convertView != null) {
+                textView = (TextView) convertView;
+            } else {
+                textView = (TextView) mInflater.inflate(
+                        getItemType(position).mLayoutRes, parent, false);
+            }
+            MenuItem item = getItem(position);
+            textView.setText(item.getTitle());
+            if (textView instanceof Checkable) {
+                ((Checkable) textView).setChecked(item.isChecked());
+            }
+            Drawable icon = item.getIcon();
+            if (icon != null) {
+                int iconWidth = icon.getIntrinsicWidth();
+                int iconHeight = icon.getIntrinsicHeight();
+                if (iconWidth < 0 || iconHeight < 0) {
+                    iconWidth = iconHeight = mPopupIconDefaultSize;
+                }
+                icon.setBounds(0, 0, iconWidth, iconHeight);
+                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1) {
+                    textView.setCompoundDrawables(icon, null, null, null);
+                } else {
+                    textView.setCompoundDrawablesRelative(icon, null, null, null);
+                }
+            }
+            textView.setPadding(mPopupPadding, position > 0 ?
+                    mPopupListDividerHeight : mPopupPadding, mPopupPadding,
+                    position < getCount() - 1 ? mPopupListDividerHeight : mPopupPadding);
+            return textView;
+        }
+    }
+}
```

Comment:
@hanningni: Anchoring to different position can be done from the client code, there is no need to change the API for that. What I actually needed to do was to support a custom background and adapter, that automatically expands top-level groups (thus enabling the header). This allows us to define our menus in standard menu resources instead of having to create a custom adapter or data structure in the code. I'll add these details on the comment at the top if you want, and I can move it to a custom `.popup` package.